src/System.Net.Primitives/src/System/Net/SocketException.Unix.cs

```diff
@@ -25,9 +25,18 @@ private SocketException(Interop.ErrorInfo error) : this(SocketErrorPal.GetSocket
 
         private static int GetNativeErrorForSocketError(SocketError error)
         {
-            return error != SocketError.SocketError ?
-                SocketErrorPal.GetNativeErrorForSocketError(error).Info().RawErrno :
-                (int)error;
+            int nativeErr = (int)error;
+            if (error != SocketError.SocketError)
+            {
+                Interop.Error interopErr = SocketErrorPal.GetNativeErrorForSocketError(error);
+                // If an interopErr was not found, then don't invoke Info().RawErrno as that will fail with assert.
+                if (nativeErr != (int)interopErr)
```

Comment:
That is not possible today as the 'interop' errors have a range (by design) that does not overlap with the SocketErrors. However I like your suggestion so will include that.